data_catalog_backend/routes/admin/category_routes.py

```diff
@@ -34,3 +35,26 @@ async def add_category(
     except Exception as e:
         logger.error(e)
         raise HTTPException(status_code=500, detail=str(e))
+
+
+@router.delete(
+    "/{id}",
+    description="Delete a category",
+    response_model=CategoryResponse,
+    response_model_exclude_none=True,
+    tags=["admin"],
+)
+async def delete_category(
+    category_id: uuid.UUID,
```

Comment:
This could potentially fail at runtime as the path-parameter in the decorator is named `{id}` but function parameter is called `category_id`. They should be named the same at least for readability of the code.